enarx-keep/shims/shim-sev/src/syscall.rs

```diff
@@ -524,4 +540,150 @@ impl Handler {
+                    0,
+                    match self.a {
+                        0 => 0x19,
+                        _ => 0xc,
+                    },
+                );
+                p.st_ino = 3;
+                p.st_mode = libc::S_IFIFO | 0o600;
+                p.st_nlink = 1;
+                p.st_uid = 1000;
+                p.st_gid = 5;
+                p.st_blksize = 4096;
+                p.st_blocks = 0;
+                p.st_rdev = makedev(0x88, 0);
+                p.st_size = 0;
+
+                p.st_atime = 1_579_507_218 /* 2020-01-21T11:45:08.467721685+0100 */;
+                p.st_atime_nsec = 0;
+                p.st_mtime = 1_579_507_218 /* 2020-01-21T11:45:07.467721685+0100 */;
+                p.st_mtime_nsec = 0;
+                p.st_ctime = 1_579_507_218 /* 2020-01-20T09:00:18.467721685+0100 */;
+                p.st_ctime_nsec = 0;
+
+                unsafe {
+                    (self.b as *mut libc::stat).write_volatile(p);
+                }
+
+                eprintln!("SC> fstat({}, {{st_dev=makedev(0, 0x19), st_ino=3, st_mode=S_IFIFO|0600,\
+                 st_nlink=1, st_uid=1000, st_gid=5, st_blksize=4096, st_blocks=0, st_size=0,\
+                  st_rdev=makedev(0x88, 0), st_atime=1579507218 /* 2020-01-21T11:45:08.467721685+0100 */,\
+                   st_atime_nsec=0, st_mtime=1579507218 /* 2020-01-21T11:45:08.467721685+0100 */,\
+                    st_mtime_nsec=0, st_ctime=1579507218 /* 2020-01-21T11:45:08.467721685+0100 */,\
+                     st_ctime_nsec=0}}) = 0", self.a);
+                Ok(0)
+            }
+            _ => Err(libc::EBADF),
+        }
+    }
+
+    pub fn fcntl(&self) -> Result<usize, libc::c_int> {
+        match (self.a as i32, self.b as i32) {
+            (libc::STDIN_FILENO, libc::F_GETFL) => {
+                eprintln!(
+                    "SC> fcntl({}, F_GETFD) = 0x402 (flags O_RDWR|O_APPEND)",
+                    self.a
+                );
+                Ok((libc::O_RDWR | libc::O_APPEND) as _)
+            }
+            (libc::STDOUT_FILENO, libc::F_GETFL) | (libc::STDERR_FILENO, libc::F_GETFL) => {
+                eprintln!("SC> fcntl({}, F_GETFD) = 0x1 (flags O_WRONLY)", self.a);
+                Ok(libc::O_WRONLY as _)
+            }
+            (libc::STDIN_FILENO, _) | (libc::STDOUT_FILENO, _) | (libc::STDERR_FILENO, _) => {
+                eprintln!("SC> fcntl({}, {}) = -EINVAL", self.a, self.b);
+                Err(libc::EINVAL)
+            }
+            (_, _) => {
+                eprintln!("SC> fcntl({}, {}) = -EBADFD", self.a, self.b);
+                Err(libc::EBADFD)
+            }
+        }
+    }
+}
+
+/// Convert an unsigned slice to a signed slice
+pub trait TrySigned<T>: Sized {
+    /// The type returned in the event of a conversion error.
+    type Error;
+
+    /// Performs the conversion.
+    fn try_signed(value: &'static [T]) -> Result<Self, Self::Error>;
+}
+
+impl TrySigned<u8> for &'static [i8] {
```

Comment:
Care to explain why those `'static`s are needed?

Reply:
right.. they are not.